src/DDTrace/Util/TryCatchFinally.php

```diff
@@ -45,6 +45,47 @@ public static function executePublicMethod(Scope $scope, $instance, $method, arr
         return $result;
     }
 
+    /**
+     * PHP 5.4 compatible try-catch-finally to execute instance methods.
+     *
+     * @param Scope $scope
+     * @param mixed $instance
+     * @param string $method
+     * @param array $args
+     * @param \Closure $afterResult
+     * @return mixed|null
+     * @throws \Exception
+     */
+    public static function executeAnyMethod(Scope $scope, $instance, $callingScope, $method, array $args = [], $afterResult = null)
+    {
+        $thrown = null;
+        $result = null;
+        /** @var SpanInterface $span */
+        $span = $scope->getSpan();
+        $call_user_func_array = function ($fn, $args) {
+            return call_user_func_array($fn, $args);
+        };
+
+        $instance_call_user_func_array->bindTo($instance, $callingScope);
+
+        try {
+            $result = $instance_call_user_func_array([$instance, $method], $args);
```

Comment:
The only way I could get this to work was by doing something like this:
```suggestion
            $result = call_user_func(\Closure::bind(
                function () use ($instance, $method, $args) {
                    return call_user_func_array([$instance, $method], $args);
                },
                null,
                $instance
            ));
```
Then we can kill the callback and the `bindTo()` lines above. We can also remove the `$callingScope` param. :)

Reply:
Thanks - Good call! :)